src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2545,6 +2545,12 @@ IntersectPlacementList(List *lhsPlacementList, List *rhsPlacementList)
 						WORKER_LENGTH) == 0)
 			{
 				placementList = lappend(placementList, rhsPlacement);
+
```

Comment:
You can think the change here as following:
```
Previously: 
First iteration: lhs: A,A   rhs:A,A result: A,A,A,A
Second iteration: lhs: A,A,A,A   rhs:A,A result: A,A,A,A,A,A,A,A
...

After the fix:
First iteration: lhs: A,A   rhs:A,A result: A,A
Second iteration: lhs: A,A   rhs:A,A result: A,A
```

As far as I can see, this change doesn't have any effect if `A`s are distinct. It'll always produce the same result.

Also, this fix aligns with the name of the function.